Development

keshav-c opened the pull request:
Awaiting review comments.

In the newsweek page the column heights in the large screens are the same. So the side columns which are smaller, scroll independently till we scroll past a point.

I haven't been able to figure out, how to recreate this effect. I hope you can help me in this area.

keshav-c committed:
Initial folder structure

keshav-c committed:
Add head

keshav-c committed:
Add outer divs for mobile first layout

keshav-c committed:
Add images, html containers

keshav-c committed:
Add rows for layout

keshav-c committed:
Add contents for top, more, feat and op

keshav-c committed:
Add latest news, mag, editors pick and slides sections

keshav-c committed:
Add magazine sections and logo image

keshav-c committed:
Add footer

keshav-c committed:
Replace local bootstrap links with CDN versions

keshav-c committed:
Add header nav

keshav-c committed:
Separate navbar into sticky and responsive

keshav-c committed:
Fix nav dropdown order, refactor css

Replace CSS with Bootstrap classes in HTML

keshav-c committed:
Make expanded nav take full width

keshav-c committed:
Adjust width of expanded nav items

keshav-c committed:
Perform validator corrections

keshav-c committed:
Make stickler recommended corrections, remove reset.css

keshav-c committed:
Remove reset.css from repo

maelfosso commented:
Hello @keshav-c 

Nice representation of that site.
There are some things to correct:

- [ ] *The padding*: You could see that all the content is surrounding from left to right by an empty space

- [ ] *The header*: The background color of the header is redder than that and please increase the height of that header. 

- [ ] *The header content*:  The date at left and the button at right are at the baseline of the header content. There are at the baseline of `NEWSWEEK`. I note that you use an image to represent `Newsweek`. If it possible please use the same thing they use. He will be easy for you to align the header content.

- [ ] *The menu*: The menu is below the header.

- [x] *The menu element*: When I hover the menu element, the background becomes black. It's not the same effect as in the original website where hovering a menu element color the text in red, the same color than the background header. 

- [ ]  *The menu content*: Also there is a padding, a space surrounding the menu content. That makes separating bar clearly visible and distinguishable.

- [ ] *The new header*: The new header that appears when scrolling down, has a smaller height.

- [ ] *The first section below the header*: The content of that section is correctly aligned with the menu, please correct it. Also, please your content doesn't have any effect neither when we hover the title (red line must underline it) nor the title of the aside content element doesn't become red when we hover each of them.  You should also try to use the same font-size than the one use in the original website.

- [ ] *Image Circle*: Please review your usage of `rounded-circle` on *img*. The image can be better than that.

- [ ] *Latest news section*: The image should all have the same size. Your first image bigger than the other ones. Increase the space between the image and the content, please. 

- [ ] *Space between two boxes*: There is always a space between two boxes, please take it in consideration. At *Feature Stories* for example, you stack everything thing without putting space. Think to *margin* there.

- [ ] *In Magazine, Editor Pick's, Featured Slideshow, Title*: You forgot to put the icon before the title.

- [ ] *In Magazine content*: Please review the size of your image. There should be higher than that.

- [ ] *Footer*: Please review the size of then element in the footer

- [ ] **Responsive Design at size** : Please review the margin. By solving problems mentioned above many will be automatically solved in responsive design view


When you finish it please submit a new code review request by mentioning me.
Thank you.

keshav-c committed:
Changed some image sizes

keshav-c committed:
Correct spacing

Add margin top to all different divs of the page. Remove border,
horizontal padding from the cards in the top half of the page.

keshav-c committed:
Add icon, more spacing changes

Add the small newsweek icon. Add vertical spacing between articles
in the featured section and also between the major sections of the
magazine.

keshav-c committed:
Adjust header spacing

Crop toggler icon for more appropriate sizing. Change horizontal
padding for various header elements for better layout on smaller
screens. Adjust vertical spacing for baselining text elements in
the header.

keshav-c committed:
Add linter suggested corrections

keshav-c commented:
These are the things I have done:

1. Corrected image sizes in the opinion section and the first image in the latest news section.
2. Added top margins for articles in the featured articles section
3. Added margins between different sections of the page to improve spacing appearance.
4. Added the Newsweek icon in places where it was missing
5. Adjust spacing, line height and padding in the header, so the toggler icon doesn't wrap below header on smaller screens.

These are the things I cannot do:

1. Putting the header in the central container breaks responsiveness and also breaks the appearance of the top non-scrolling header. 
2. Also the scrolling header is not contained in the container. so it breaks appearance if the scroll bar is at the top of the screen.
3. Adding margin breaks responsive layout and it pushes the sections to the next row even in wider resolutions.

For the changes I have not been able to implement it _will be better if you can sit with me in a session_, to help out. 

If you are unable, to take the time please consider approving the project since most of the changes you have asked are cosmetic and are **not** a core requirement of the project, which is **to demonstrate the use of Bootstrap to clone the page**.

keshav-c committed:
Make wider screen header, have a color separation between menus

keshav-c committed:
Add bg color to page, put footer in container

maelfosso commented on index.html:
```diff
@@ -0,0 +1,726 @@
+                </ul>
+            </div>
+        </nav>
+        <nav class="navbar navbar-dark bg-danger fixed-top px-1">
```
Set the background of that **nav** to `red`

keshav-c replied:
ok. I have made the change. But there is no difference between those 2. On the other hand, since this is a bootstrap project, there should be less emphasis of specifying these details through the CSS. The point should be to write as less CSS as possible, and making use of the Bootstrap classes.

maelfosso commented on index.html:
```diff
@@ -0,0 +1,726 @@
+        <link rel="stylesheet" href="./styles/main.css">
+    </head>
+    <body>
+        <nav class="responsive-nav navbar navbar-expand-lg navbar-dark bg-danger flex-wrap pb-lg-0 px-1">
```
Wrap the two `nav` into a `header`
The content of that can be wrap into a `<div class="container">` or `<div class="container-fluid">`

maelfosso replied:
Why did you use `responsive-nav` and `bg-danger` instead of setting `background-color:red` ?

keshav-c replied:
I have made header and footer changes.

As I mentioned above, there is no difference between those 2. Even if danger and red are slightly different shades,  they are only colors which don't make a difference to the layout. On the other hand, since this is a bootstrap project, there should be less emphasis of specifying these details through the CSS. The point should be to write as less CSS as possible, and making use of the Bootstrap classes.

maelfosso commented on index.html:
```diff
@@ -0,0 +1,726 @@
+                <a class="nav-item nav-link text-white font-weight-bold pr-1 pr-sm-1" href="#">SIGN IN</a>
+                <a class="nav-item nav-link bg-white text-danger font-weight-bold px-1 px-sm-1" href="#">SUBSCRIBE</a>
+            </div>
+            <div class="navbar-nav">
```
You don't need that.

keshav-c replied:
Wrapped the links and button in a single div as suggested.

maelfosso commented on index.html:
```diff
@@ -0,0 +1,726 @@
+        <nav class="responsive-nav navbar navbar-expand-lg navbar-dark bg-danger flex-wrap pb-lg-0 px-1">
+            <span class="navbar-text text-white d-none d-lg-inline">Current date</span>
+            <a href="#" class="navbar-brand d-lg-none pt-1 pb-1"><img class="small" src="./images/logo-white.png" alt="logo"></a>
+            <a href="#" class="navbar-brand d-none d-lg-inline flex-lg-grow-1 text-center pt-1 pb-1"><img class="large" src="./images/logo-white.png" alt="logo"></a>
```
Why did you put something with a `d-none` here ?

keshav-c replied:
Because that should not be rendered in smaller width screens. The date string is not visible on header on smaller width screens.

maelfosso commented on index.html:
```diff
@@ -0,0 +1,726 @@
+                    </div>
+                </div>
+            </div>
+            <div class="row magazine-section">
```
Let's take this last part before the row for example.
```
<section>
    <h3>SECTION F</h3>
    <div class="d-flex">
       .....
    </div>
</section>
```

keshav-c replied:
d-flex is not needed. I have achieved the same effect using grid. The appearance looks fine to me. Even if I added a container that uses `d-flex` I will still have to give a width to each article using grid sizing (`col-md-4` etc). 

Explain, why you want this to be changed first. That is, how is the behaviour of my code different from desired? And how would this change help?

maelfosso commented on index.html:
```diff
@@ -0,0 +1,726 @@
+                </ul>
+            </div>
+        </nav>
+        <div class="container bg-white">
```
Change that part, please

```html
<main>
    <div class="row-fluid">
        <div class=""></div>
        <div class=""></div>
    </div>
</main>
````

keshav-c replied:
I don't feel comfortable making this change, because the responsive behaviour of my page is already correct. I have checked my page multiple times (both locally and online), and the responsive behaviour is correct for all different screen width breakpoints.

If you see something that is wrong, point it out to me and then I will make the change.

maelfosso commented:
Hello,

Good ameliorations

There are few things to do. 

- [ ] **Second navbar _fixed top_** : Set the background to `red`, He will so have the same background color than the first one

- [ ] **The menu**: The menu, the hover effect please improve it.

- [ ] **Main content**: Wrap the two `<div class="container bg-white"> into a `main` tag nested by a `<div class="container-fluid">`. Then you remove the two `container bg-white`. 

- [ ] **Section part**: Please use `section` for different part of the main content. That will help you to design them correct

- [ ] **Footer**: Remove the **footer** from where you put it. Make the footer come directly after the `main` tag

Take those points in consideration, please. In case of problem, let's meet together on Zoom.

keshav-c committed:
changed nav background color

keshav-c committed:
Mod header and footer, with appropriate container classes

keshav-c committed:
Consolidate navbar links in a single container

keshav-c committed:
Change header hover effects

keshav-c commented:
> Hello,
> 
> Good ameliorations
> 
> There are few things to do.
> 
> * [ ]  **Second navbar _fixed top_** : Set the background to `red`, He will so have the same background color than the first one
> * [ ]  **The menu**: The menu, the hover effect please improve it.
> * [ ]  **Main content**: Wrap the two `<div class="container bg-white"> into a `main`tag nested by a``. Then you remove the two `container bg-white`.
> * [ ]  **Section part**: Please use `section` for different part of the main content. That will help you to design them correct
> * [ ]  **Footer**: Remove the **footer** from where you put it. Make the footer come directly after the `main` tag
> 
> Take those points in consideration, please. In case of problem, let's meet together on Zoom.

I have changed the hover effect, nav background (check in _main.css_ line 145) and the footer.

The rest of the changes seem will not change appearance in anyway, so I have left those as they are.

Awaiting further comments.

keshav-c committed:
Add left margin to header date string

maelfosso commented:
# Good job 

You fill the completion form

keshav-c commented:
Thanks.
